blocks/components/editable/index.js

```diff
@@ -2,32 +2,64 @@
  * External dependencies
  */
 import classnames from 'classnames';
-import { forEach, last } from 'lodash';
+import { last } from 'lodash';
 import { Parser as HtmlToReactParser } from 'html-to-react';
+import { Fill } from 'react-slot-fill';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
 
+ // TODO: We mustn't import by relative path traversing from blocks to editor
+ // as we're doing here; instead, we should consider a common components path.
+import Toolbar from '../../../editor/components/toolbar';
+
 const htmlToReactParser = new HtmlToReactParser();
 const formatMap = {
 	strong: 'bold',
 	em: 'italic',
 	del: 'strikethrough'
 };
 
+const formattingControls = [
+	{
+		icon: 'editor-bold',
+		title: wp.i18n.__( 'Bold' ),
+		format: 'bold'
+	},
+	{
+		icon: 'editor-italic',
+		title: wp.i18n.__( 'Italic' ),
+		format: 'italic'
+	},
+	{
+		icon: 'editor-strikethrough',
+		title: wp.i18n.__( 'Strikethrough' ),
+		format: 'strikethrough'
+	}
+];
+
 export default class Editable extends wp.element.Component {
 	constructor() {
 		super( ...arguments );
+
 		this.onInit = this.onInit.bind( this );
 		this.onSetup = this.onSetup.bind( this );
 		this.onChange = this.onChange.bind( this );
 		this.onNewBlock = this.onNewBlock.bind( this );
 		this.bindNode = this.bindNode.bind( this );
 		this.onFocus = this.onFocus.bind( this );
+		this.onFocusOut = this.onFocusOut.bind( this );
 		this.onNodeChange = this.onNodeChange.bind( this );
+
 		this.formats = {};
+		this.nodes = {};
+
+		this.state = {
+			isFocused: false,
```

Comment:
Shouldn't we drop the `isFocued` here and rely on `!! this.props.focus` instead?